src/analyses/threadCreateEdges.ml

```diff
@@ -0,0 +1,49 @@
+(* threadCreateEdges: Tracks which create edges have been encountered since the beginning of the current function *)
```

Comment:
I'm a bit puzzled by what this is really doing. This is just using the thread ID domain as its local domain and delegating to the thread ID analysis. It's not tracking any additional thread creation edges, but simply reusing the set from within the thread ID domain. And then the thread ID analysis queries this for the set.

Why this roundabout way? if the data already exists inside the thread ID analysis domain and is tracked by the thread ID analysis, why can't it just directly use the set itself?

Reply:
Could the existing created set in the thread ID domain simply be replaced with a pair of sets then? One tracking the existing thing and another the one which starts out empty.
Or maybe just have the new analysis's local value be added as the third one to thread ID analysis?

Reply:
One could for sure do that. The question just is if this does not cause overhead in the default configuration? Currently we don't have any overhead if these options are disabled.
Also, if I am context-sensitive in the set of encountered thread creation edges, this additional component is not helpful.

Reply:
I suspect the overhead of an extra analysis has more overhead than an extra component in an existing analysis, but that all depends on which is the default. Since this analysis isn't the default, then it might be slightly better this way.

Although I'm still wondering if the query is necessary. Couldn't the new analysis simply be context-sensitive instead of making threadid query this one for what to use in its context?